techniques/system/common/1.0/promises.st

```diff
@@ -635,6 +638,52 @@ bundle agent check_rsyslog_version {
 
 }
 
+#############################################################
+# Remove syslog configurations that Rudder may have applied #
+#############################################################
+
+bundle agent remove_rudder_syslog_configuration {
+
+  vars:
+
+      "syslogng_delete_lines_patterns" slist => {
+        "\s*# Autogenerated by rudder, do not edit.*",
+        ".*destination\(rudder_loghost\);.*",
+      };
+
+      "syslog_delete_lines_patterns" slist => {
+        "\s*# Rudder specific logging parameters.*",
+        "local6.notice\s+@${server_info.cfserved}",
+      };
+
+  classes:
+
+      "syslogng" expression => fileexists("${check_log_system.syslogng_conffile}");
+      "syslogd"  expression => fileexists("${check_log_system.syslog_conffile}");
+
+  files:
+      "${check_log_system.rsyslog_rudder_conffile}"
+        delete         => tidy,
+        classes        => classes_generic("remove_rudder_syslog_configuration_remove_files");
```

Comment:
This file removal could be only done if we detect rsyslog (like the 2 following blocks), but since it is most definitely our file (named rudder-agent.conf) and CFEngine considers this is a success, we might as well do it every time, just in case a node had a rsyslog server on it once - a good chance to clear up, and less code to maintain.